specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-01/managedClusters.json

```diff
@@ -2078,6 +1947,25 @@
       "type": "string",
       "description": "Client Api Version.",
       "x-ms-parameter-location": "client"
+    },
+    "ResourceGroupNameParameter": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "minLength": 1,
+      "description": "The name of the resource group.",
+      "x-ms-parameter-location": "method"
+    },
+    "ResourceNameParameter": {
+      "name": "resourceName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 63,
```

Comment:
@haitch won't this have a breaking change for clients? Was service failing earlier due to RG name greater than 63? Shouldn't this be applicable to all RPs and not just yours?

Reply:
this is specific for AKS managedCluster resourceName.

in azure Portal we have clusterName limit to 3-31 characters
in azure CLI we have limit to 1-63.

This is to align with CLI, portal will be updated shortly.

Reply:
our current client doesn't limit the resource name, but it will fail at server side, so this is not a regression, and not a breaking change.

Reply:
where is the regex pattern?

Reply:
Do we need to wait for a regex pattern concern? Or is this ready to be merged?

Reply:
commit 414a3de already append regex